php/php.editor/src/org/netbeans/modules/php/editor/csl/SemanticAnalysis.java

```diff
@@ -595,23 +596,27 @@ public void visit(ClassInstanceCreation node) {
                 // to avoid recognizing $this as an instance of an anonymous class
                 scan(node.ctorParams());
                 addToPath(node);
+                // GH-5551 keep original type info to scnan parent blocks
+                TypeInfo originalTypeInfo = typeInfo;
                 typeInfo = new ClassInstanceCreationTypeInfo(node);
                 scan(node.getAttributes());
                 scan(node.getSuperClass());
                 scan(node.getInterfaces());
-                needToScan = new ArrayList<>();
+                needToScan.put(typeInfo, new ArrayList<>());
                 Block body = node.getBody();
                 if (body != null) {
                     body.accept(this);
 
                     // find all usages in the method bodies
-                    while (!needToScan.isEmpty()) {
-                        Block block = needToScan.remove(0);
+                    while (!needToScan.get(typeInfo).isEmpty()) {
+                        Block block = needToScan.get(typeInfo).remove(0);
```

Comment:
_Note:_ Since you are removing from the first index, perhaps `ArtrayList` is not the best list implementation for it. So, perhaps `LinkedList` would be better here (although, personally, I don't like it :).

Reply:
Use for loop?
```java
                for (Block block : needToScan.get(typeInfo)) {
                    block.accept(this);
                }
                needToScan.get(typeInfo).clear();

```

Reply:
Fixed. Thank you!